cdap-app-templates/cdap-etl/cdap-data-streams/src/test/java/co/cask/cdap/datastreams/DataStreamsSparkSinkTest.java

```diff
@@ -134,5 +135,6 @@ public Boolean call() throws Exception {
 
     sparkManager.stop();
     sparkManager.waitForStatus(false, 10, 1);
+    sparkManager.waitForRun(ProgramRunStatus.KILLED, 10, TimeUnit.SECONDS);
```

Comment:
So, if we don't wait for the run record to be persisted in the store, it will try to do this in the next test that it runs, and in the next test, it will persist KILLED, which would throw an exception because there is no starting/running record.

In my branch that switches to TMS, I've had to do this to many tests as the tests do not correctly wait for the record to be persisted.